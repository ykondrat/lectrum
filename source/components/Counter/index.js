import React from 'react';
import { number } from 'prop-types';
import Styles from './styles.m.css';

const Counter = (props) => (
    <section className = { Styles.counter }>Posts count: { props.count }</section>
)

Counter.propType = {
    count: number.isRequired
}

Counter.defaultProps = {
    count: 0
}

export default Counter;
