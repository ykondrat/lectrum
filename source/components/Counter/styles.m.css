.counter {
    display: flex;
    justify-content: center;
    position: relative;
    color: var(--paletteColor7);
    margin-bottom: 10px;
}
