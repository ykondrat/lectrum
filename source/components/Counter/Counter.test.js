import Counter from './';
import dom from 'react-test-renderer';

const renderTree = dom.create(<Counter count = { 5 } />).toJSON();

describe('Counter component', () => {
    test('Counter component should correspond to its snapshot', () => {
        expect(renderTree).toMatchSnapshot();
    });
});
