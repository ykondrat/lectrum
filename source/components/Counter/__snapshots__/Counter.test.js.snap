// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Counter component should correspond to its snapshot 1`] = `
<section
  className="counter"
>
  Posts count: 
  5
</section>
`;
